### PR TITLE
feat: PRO-2436 sdk function to get all modules from etherspot-wallet

### PR DESCRIPTION
**Pull request by kanthgithub**
SDK functions to get list of all modules installed in the etherspot wallet & Uninstall module data generation

## Description

1. add new functions to get all executors, validators, active-hook from paginated functions of reference implementation in 
-   ModuleManager
-   HookManager
 
2. added SDK helper functions to generate deInitData for `uninstallModule`

## Types of changes

- [ X] New feature (non-breaking change which adds functionality)
- [ X] Documentation Update

### CODE REVIEW

File: src/sdk/base/EtherspotWalletAPI.ts [hunk 1]
```diff
@@ -49,10 +63,18 @@ export class EtherspotWalletAPI extends BaseAccountAPI {
...
+
+  async isModuleInstalled(moduleTypeId: MODULE_TYPE, module: string): Promise<boolean> {
+    const accountContract = EtherspotWallet7579__factory.connect(await this.getAccountAddress(), this.provider);
+    return await accountContract.isModuleInstalled(moduleTypeId, module, '0x');
   }
 
   async installModule(moduleTypeId: MODULE_TYPE, module: string, initData = '0x'): Promise<string> {
+    const accountAddress = await this.getAccountAddress();
     const accountContract = EtherspotWallet7579__factory.connect(await this.getAccountAddress(), this.provider);
+    console.log(`accountContractAddress: ${accountAddress} \n moduleTypeId: ${moduleTypeId} \n module: ${module} \n initData: ${initData}`);
```
Comment (ch4r10t33r) [inline src/sdk/base/EtherspotWalletAPI.ts:77]: Please remove console.logs before requesting review.

File: src/sdk/base/EtherspotWalletAPI.ts [hunk 2]
```diff
@@ -69,8 +91,178 @@ export class EtherspotWalletAPI extends BaseAccountAPI {
...
+    // this is applicable only for Executor and Validator modules
+    // if the module type is not Executor or Validator, throw an error
+    if (moduleTypeId !== MODULE_TYPE.EXECUTOR && moduleTypeId !== MODULE_TYPE.VALIDATOR) {
+      throw new Error("Unsupported module type");
+    }
+
+    // Get the previous address in the list
+    const previousAddress = await this.getPreviousAddressInSentinelList(module, moduleTypeId);
+
+    console.log(`Previous address: ${previousAddress}`);
```
Comment (ch4r10t33r) [inline src/sdk/base/EtherspotWalletAPI.ts:195]: Please see above.

File: src/sdk/base/EtherspotWalletAPI.ts [hunk 3]
```diff
@@ -69,8 +91,178 @@ export class EtherspotWalletAPI extends BaseAccountAPI {
...
+        break;
+      }
+    } while (true);
+
+    return totalExecutors;
+  }
+
+  // Adjusting the function to be generic for handling different module types
+  async getPreviousAddressInSentinelList(targetAddress: string, moduleTypeId: MODULE_TYPE): Promise<string> {
+    let pageSize = 50;
```
Comment (ch4r10t33r) [inline src/sdk/base/EtherspotWalletAPI.ts:124]: Should we pass this as a configurable parameter (with default value as 50)?

File: src/sdk/base/EtherspotWalletAPI.ts [hunk 4]
```diff
@@ -69,8 +91,178 @@ export class EtherspotWalletAPI extends BaseAccountAPI {
     return accountContract.interface.encodeFunctionData('uninstallModule', [moduleTypeId, module, deinitData]);
   }
 
+  // Function to get all executors
+  async getAllExecutors(): Promise<string[]> {
+    // Assuming getExecutorsPaginated is a contract function that can be called
+    // Initialize variables
+    let pageSize = 50;
```
Comment (ch4r10t33r) [inline src/sdk/base/EtherspotWalletAPI.ts:98]: Should we pass this as a configurable parameter (with default value as 50)?

File: src/sdk/base/EtherspotWalletAPI.ts [hunk 5]
```diff
@@ -69,8 +91,178 @@ export class EtherspotWalletAPI extends BaseAccountAPI {
...
+      validators,
+      executors,
+      hook,
+      fallbacks
+    };
+  }
+
+
   async checkAccountAddress(address: string): Promise<void> {
+    console.log(`inside checkAccountAddress for: ${address}`);
```
Comment (ch4r10t33r) [inline src/sdk/base/EtherspotWalletAPI.ts:263]: Please remove console.log

**Review by ch4r10t33r — CHANGES_REQUESTED** [review]
Few minor comments.

File: src/sdk/base/EtherspotWalletAPI.ts (hunk 5 above)
Comment (kanthgithub) [reply src/sdk/base/EtherspotWalletAPI.ts]: oops forgot to push removed changes while raising PR. removed it now

File: src/sdk/base/EtherspotWalletAPI.ts (hunk 3 above)
Comment (kanthgithub) [reply src/sdk/base/EtherspotWalletAPI.ts]: updated with usage of DEFAULT_PAGE_SIZE and caller can override with the value they wish to paginate with

File: src/sdk/base/EtherspotWalletAPI.ts (hunk 4 above)
Comment (kanthgithub) [reply src/sdk/base/EtherspotWalletAPI.ts]: updated with usage of DEFAULT_PAGE_SIZE and caller can override with the value they wish to paginate with

**Review by ch4r10t33r — APPROVED** [review]
LGTM

**Pull request merged**